### ISSUE TITLE
[Due for payment 2025-03-03] [$250] Track expense - Back button returns to expense report after submitting track expense

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#52189]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.59-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N/A - new feature, doesn't exist in prod
**Email or phone of affected tester (no customers):** applausetester+zm27@applause.expensifail.com
**Issue reported by:** Applause Internal Team
## Action Performed:
1. Go to staging.new.expensify.com
2.Go to self DM.
3. Track a manual expense.
4. Go to expense report.
5. Tap on the header.
6. Tap Submit to someone.
7. Select any user and submit it.
8. Tap app back button.

## Expected Result:
App will return to self DM because the expense is submitted and therefore should be removed from the navigation path when returning to the previous page (self DM).

## Actual Result:
App returns to the track expense. It only returns to self DM on second tap on the back button.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/c6030838-9e03-4ae3-b352-661a1ff48514

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021854844368006624450</li>
        <li>Upwork Job ID: 1854844368006624450</li>
        <li>Last Price Increase: 2024-11-15</li>
<li>Automatic offers: </li>
<ul>
<li>daledah | Contributor | 105023189</li>
</ul></li>
    </ul>
</details>

**Comment by Gonals** [comment]
This seems minor enough no to block deploy on (today is Friday, anyways). What do you think, @Julesssss?

**Comment by Julesssss** [comment]
I agree it's not major, but lets at least locate the regression first. Deploy won't be happening today so we have a bit of time.

**Comment by Julesssss** [comment]
@Gonals it sounds possibly related to [this other issue](https://github.com/Expensify/App/issues/52240)

**Comment by mountiny** [comment]
Whoever identifies the offending PR gets $50 and we can revert

**Comment by Gonals** [comment]
@eVoloshchak quick C+ review of the proposal here? 😁

**Comment by zanyrenney** [comment]
@twisterdotcom seeing as you're back and this wasn't daily, so I haven't contributed, going to unassign myself!

**Comment by mountiny** [comment]
I believe we agreed to demote this in Slack, so doing so

@Gonals and @eVoloshchak can handle the review process

**Comment by eVoloshchak** [comment]
@daledah's [proposal](https://github.com/Expensify/App/issues/52189#issuecomment-2467421420) looks good to me!
The same bug happens when selecting "Categorize it" and "Share it with my accountant" due to the same root cause, should also be resolved by the fix, let's add the tests for these cases too

🎀👀🎀 C+ reviewed!

**Comment by twisterdotcom** [comment]
Starring @Gonals

**Comment by twisterdotcom** [comment]
How is this going @daledah?

**Comment by twisterdotcom** [comment]
Back from break. How is the review going @eVoloshchak? Looks like we're waiting on you here now @daledah: https://github.com/Expensify/App/pull/53065#pullrequestreview-2524989069

**Comment by twisterdotcom** [comment]
Discussion ongoing in the PR: https://github.com/Expensify/App/pull/53065

**Comment by eVoloshchak** [comment]
The PR has been merged, but this will need a follow-up PR due to ongoing navigation changes (https://github.com/Expensify/App/pull/53065#issuecomment-2670964021)

**Comment by twisterdotcom** [comment]
Okay, should we pay out now, or wait?

**Comment by eVoloshchak** [comment]
@twisterdotcom, still duscussing what could be done [here](https://github.com/Expensify/App/pull/53065), let's wait for a bit

**Comment by twisterdotcom** [comment]
As @eVoloshchak says we're holding on this: https://github.com/Expensify/App/pull/53065

**Comment by eVoloshchak** [comment]
Yes, waiting for a response from @daledah in https://github.com/Expensify/App/pull/53065#issuecomment-2709756504

**Comment by twisterdotcom** [comment]
Bump on this then @daledah.

**Comment by eVoloshchak** [comment]
@twisterdotcom, from https://github.com/Expensify/App/pull/53065#issuecomment-2728528185
> After https://github.com/Expensify/App/pull/49539 is merged the navigation flow is a lot different now, and I can't find a solution anymore after quite some days investigating. Please consider looking for another contributor who is more familiar with the new navigation flow to fix the issue.

I agree with @daledah, this was working properly before the major navigation overhaul, regression wasn't caused by this PR.
Not sure what should be our next steps here, should I report the new issue (behavior for browser back button being different from hardware back button) on Slack?

**Comment by twisterdotcom** [comment]
Yes, let's just report it as a new bug. We should still pay this out though right if a PR was raised and reviewed?

**Comment by twisterdotcom** [comment]
@eVoloshchak and @Gonals - do you agree we should pay this out?

**Comment by eVoloshchak** [comment]
I agree, the work was done by both parties and the fix was working before major navigation PR was merged. 
The difference in behavior between the browser back button and in-app back button is ann app-wide issue, not related to this one. 
It looks like there is some discussion on whether it should be fixed [here](https://expensify.slack.com/archives/C08CZDJFJ77/p1742583885690369)

**Comment by twisterdotcom** [comment]
Okay, so this is basically [point 17](https://docs.google.com/document/d/1BvohU05MTaHnjOD_vwJv_aqDAirv-ChkyRnKCAvOVyQ/edit?tab=t.0#bookmark=id.wfm6lkgp0c6) of the process doc.

> If a contributor has been hired for a job and we decide to close the job before it is successfully completed
> 50% payment to contributor and C+ if the contributor was 🎀👀🎀 by a C+ or hired/assigned. 
> 100% due if contributor drafted a PR and requested a review. ✅ 
> 100% due to C+ if they reviewed the PR ✅ 

So, were really just need to pay out $250, close this issue and fix it properly in https://github.com/Expensify/App/issues/59245

**Comment by twisterdotcom** [comment]
Payment Summary:

- @eVoloshchak paid $250 via NewDot Manual Request
- @daledah paid $250 [here](https://www.upwork.com/nx/wm/offer/105023189) (Contributor)
